plugins/osdn/factory/factory.go

```diff
@@ -13,13 +13,23 @@ import (
 )
 
 // Call by higher layers to create the plugin instance
-func NewPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string) (api.OsdnPlugin, api.FilteringEndpointsConfigHandler, error) {
+func NewPlugin(pluginType string, osClient *osclient.Client, kClient *kclient.Client, hostname string, selfIP string) (api.OsdnPlugin, error) {
```

Comment:
if we're going to change the API here, then can we also split this into NewMasterPlugin() and NewNodePlugin()? (Where for now `NewMasterPlugin(pluginType, osClient, kClient)` just calls `NewNodePlugin(pluginType, osClient, kClient, "", "")`). Yeah, it doesn't have anything to do with the rest of this patch, but if we're going to be having to do synced origin/sdn changes anyway...